client/lib/upgrades/actions/purchases.js

```diff
@@ -19,6 +20,8 @@ const PURCHASES_FETCH_ERROR_MESSAGE = i18n.translate( 'There was an error retrie
 	PURCHASE_REMOVE_ERROR_MESSAGE = i18n.translate( 'There was an error removing the purchase.' );
 
 function cancelPurchase( purchaseId, onComplete ) {
+	olarkActions.setNotReady();
```

Comment:
I think we could eliminate the `OLARK_NOT_READY` and `setNotReady` actiontype/action and just call the new [`olark.fetchUserEligibility()`](https://github.com/Automattic/wp-calypso/pull/4649/files#r59305069) here.